src/super_gradients/training/models/sg_module.py

```diff
@@ -1,11 +1,12 @@
-from typing import Union
+from typing import Union, Optional, Callable
 
 from torch import nn
 
 from super_gradients.training.utils.utils import HpmStruct
+from super_gradients.module_interfaces import SupportsReplaceInputChannels
 
 
-class SgModule(nn.Module):
+class SgModule(nn.Module, SupportsReplaceInputChannels):
```

Comment:
Originally I did not have this `SupportsReplaceInputChannels `, but there are some models that are not `SgModule` but that still need to support `replace_input_channels`, like backbones and stems.
Having an external `SupportsReplaceInputChannels` solves the problem, because we can make `SgModule` inherit from it, but also any other backbone/stem that is not `SgModule`